worker/wallet.js

```diff
@@ -349,3 +355,44 @@ export async function checkPendingWithdrawals (args) {
     }
   }
 }
+
+async function handleAction ({ data: { msatsReceived, actionType, actionId, actionData }, models }) {
+  if (!actionType || !actionId) return
+
+  if (actionType === 'ITEM') {
+    // update item status from PENDING to ACTIVE
+    // and run queries which were skipped during creation
+
+    const itemId = actionId
+    const { credits: locked, cost } = actionData
+
+    await models.$transaction(async (tx) => {
+      const item = await tx.item.update({
+        where: { id: itemId },
+        data: { status: 'ACTIVE' }
+      })
+
+      await tx.user.update({
+        where: {
+          id: item.userId
+        },
+        data: {
+          msats: {
+            decrement: cost - msatsReceived - locked
+          }
+        }
+      })
+
+      // run skipped item queries
+      await tx.itemAct.create({
+        data: { msats: cost, itemId: item.id, userId: item.userId, act: 'FEE' }
+      })
+      if (item.boost > 0) {
+        await tx.$executeRaw(`SELECT item_act(${item.id}::INTEGER, ${item.userId}::INTEGER, 'BOOST'::"ItemActType", ${item.boost}::INTEGER)`)
+      }
+      if (item.maxBid) {
+        await tx.$executeRaw(`SELECT run_auction(${item.id}::INTEGER);`)
+      }
+    }, { isolationLevel: Prisma.TransactionIsolationLevel.Serializable })
+  }
+}
```

Comment:
Just a heads up (I know this is in draft and this might not be what you intend to ship): worst case, this is 5 roundtrips to the db (because it's interactive and not pipelined), and if during that time any of the modified rows are updated by another transaction, this and the other tx will fail. 

In general, the longer a serializable tx takes to run, the more likely the failure. Everywhere in the code so far, we only do one roundtrip for serializable txs (and we wrap them all in a retry in `serializable`)

Reply:
> worst case, this is 5 roundtrips to the db (because it's interactive and not pipelined) [...] Everywhere in the code so far, we only do one roundtrip for serializable txs (and we wrap them all in a retry in serializable)

Oh, I didn't realize that the `serialize` calls with an array of queries weren't multiple round trips.

But I agree with you, roundtrips are bad, especially when it's in a serialized tx. I didn't think too much about it here for now, it was just too convenient to use Prisma's API. 

But I noticed now that the only reason I use an interactive tx was to fetch the item author. I can do this outside of the tx and then use `serialize`.